### PR TITLE
git-init update and fixes

### PR DESCRIPTION
**Pull request by vdemeester**
## Proposed Changes

- check for errors or at least log them trim spaces from url : if the
- given url contains some spaces (like non-breaking spaces),
  `git-init` may fail with `protocol ' https' is not supported`. This
  fixes that.

Related to https://github.com/tektoncd/pipeline/pull/677 and https://github.com/tektoncd/pipeline/pull/750

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

### CODE REVIEW

File: cmd/git-init/main.go [hunk 1]
```diff
@@ -87,16 +91,18 @@ func main() {
 		run(logger, "git", "init")
```
Comment (chmouel) [inline cmd/git-init/main.go]: Why you do that twice?

And perhaps check if path+"./git" exist already ?

**Comment by chmouel** [comment]
Perhaps add this review too https://github.com/tektoncd/pipeline/pull/747

**Comment by zouyee** [comment]
I think `path` also need to be validated.

File: cmd/git-init/main.go (hunk 1 above)
Comment (vdemeester) [reply cmd/git-init/main.go]: @chmouel it's not on the same "branch", so still needed

File: cmd/git-init/main.go [hunk 2]
```diff
@@ -87,16 +91,18 @@ func main() {
...
 	}
 
-	run(logger, "git", "remote", "add", "origin", *url)
-	err = runOrFail(logger, "git", "fetch", "--depth=1", "--recurse-submodules=yes", "origin", *revision)
-	if err != nil {
+	trimedURL := strings.TrimSpace(*url)
+	runOrFail(logger, "git", "remote", "add", "origin", trimedURL)
+	if err = run(logger, "git", "fetch", "--depth=1", "--recurse-submodules=yes", "origin", *revision); err != nil {
```
Comment (zouyee) [inline cmd/git-init/main.go]: Could we consider using context to handler command hangs? Today we encountered a situation which the command was stuck, and there was a problem with the underlying storage.

Comment (vdemeester) [reply cmd/git-init/main.go]: @zouyee we may, with a default `timeout`, question is then, what would be a "sane" timeout ?

**Comment by ImJasonH** [comment]
Closing this as stale, please reopen if you think this is still something we should do.

**Pull request closed without merging**